src/test/java/uk/gov/pay/ledger/pact/PaymentDetailsEnteredEventQueueContractTest.java

```diff
@@ -0,0 +1,94 @@
+import uk.gov.pay.ledger.event.dao.EventDao;
+import uk.gov.pay.ledger.rule.AppWithPostgresAndSqsRule;
+import uk.gov.pay.ledger.rule.SqsTestDocker;
+import uk.gov.pay.ledger.transaction.dao.TransactionDao;
+import uk.gov.pay.ledger.transaction.entity.TransactionEntity;
+import uk.gov.pay.ledger.util.fixture.QueueEventFixture;
+
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static io.dropwizard.testing.ConfigOverride.config;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class PaymentDetailsEnteredEventQueueContractTest {
+    @Rule
+    public MessagePactProviderRule mockProvider = new MessagePactProviderRule(this);
+
+    @Rule
+    public AppWithPostgresAndSqsRule appRule = new AppWithPostgresAndSqsRule(
+            config("queueMessageReceiverConfig.backgroundProcessingEnabled", "true")
+    );
+
+    private byte[] currentMessage;
+    private String externalId = "externalId";
+    private ZonedDateTime eventDate = ZonedDateTime.parse("2018-03-12T16:25:01.123456Z");
+    private String gatewayAccountId = "gateway_account_id";
+
+    @Pact(provider = "connector", consumer = "ledger")
+    public MessagePact createPaymentDetailsEnteredEventPact(MessagePactBuilder builder) {
+        QueueEventFixture paymentDetailsEntered = QueueEventFixture.aQueueEventFixture()
+                .withResourceExternalId(externalId)
+                .withEventDate(eventDate)
+                .withGatewayAccountId(gatewayAccountId)
+                .withDefaultEventDataForEventType("PAYMENT_DETAILS_ENTERED");
+
+        Map<String, String> metadata = new HashMap<>();
+        metadata.put("contentType", "application/json");
+
+        return builder
+                .expectsToReceive("a payment details entered message")
+                .withMetadata(metadata)
+                .withContent(paymentDetailsEntered.getAsPact())
+                .toPact();
+    }
+
+    @Test
+    @PactVerification({"connector"})
+    public void test() throws Exception {
+        appRule.getSqsClient().sendMessage(SqsTestDocker.getQueueUrl("event-queue"), new String(currentMessage));
+
+        TransactionDao transactionDao = new TransactionDao(appRule.getJdbi());
+        EventDao eventDao = appRule.getJdbi().onDemand(EventDao.class);
+
+        await().atMost(1, TimeUnit.SECONDS).until(
+                () -> transactionDao.findTransactionByExternalId(externalId).isPresent()
+        );
+
+        Optional<TransactionEntity> transaction = transactionDao.findTransactionByExternalId(externalId);
+        assertThat(transaction.isPresent(), is(true));
+        assertThat(transaction.get().getExternalId(), is(externalId));
+        assertThat(transaction.get().getEmail(), is("j.doe@example.org"));
```

Comment:
Could get away with hard coded string if fixture is used as here https://github.com/alphagov/pay-ledger/blob/55cf5db6d546d996da2aeeb76d9ae5158ee6963d/src/test/java/uk/gov/pay/ledger/pact/RefundCreatedByUserEventQueueContractTest.java#L41

Reply:
I'm rather a huge fan of hardcoded strings in tests.

Reply:
Not a big issue but we are ending up with different patterns with in same code base which we might want to avoid